src/main/java/core/basesyntax/Main.java

```diff
@@ -0,0 +1,17 @@
+package core.basesyntax;
+
+public class Main {
+    public static void main(String[] args) {
+        FigureSupplier figureSupplier = new FigureSupplier();
+        Figure[] figures = new Figure[6];
+        for (int i = 0; i < figures.length / 2; i++) {
+            figures[i] = figureSupplier.getRandomFigure();
+        }
+        for (int i = figures.length / 2; i < figures.length; i++) {
+            figures[i] = figureSupplier.getRandomFigure();

```

Comment:
You are creating random figures in two separate loops that effectively do the same thing. This can be simplified into a single loop that populates the entire array with random figures. Consider merging these loops into one.